kosmorrolib/events.py

```diff
@@ -528,3 +528,88 @@ def get_events(for_date: date = date.today(), timezone: int = 0) -> [Event]:
+
+
+def search_events(
+    event_types: [EventType], end: date, start: date = date.today(), timezone: int = 0
+) -> [Event]:
+    """Calculate and return a list of events for the given time range, adjusted to a given timezone.
+
+    Find all events between January 27th, 2020 and January 29th, 2020:
+
+    >>> event_types = [EventType.OPPOSITION, EventType.CONJUNCTION, EventType.OCCULTATION, EventType.MAXIMAL_ELONGATION, EventType.APOGEE, EventType.PERIGEE, EventType.SEASON_CHANGE, EventType.LUNAR_ECLIPSE]
+    >>> search_events(event_types, date(2020, 1, 29), date(2020, 1, 27)) # doctest: +NORMALIZE_WHITESPACE
+    [<Event type=CONJUNCTION objects=[<Object type=PLANET name=VENUS />, <Object type=PLANET name=NEPTUNE />] start=2020-01-27 20:00:23.242428+00:00 end=None details=None />,
+    <Event type=CONJUNCTION objects=[<Object type=SATELLITE name=MOON />, <Object type=PLANET name=NEPTUNE />] start=2020-01-28 09:33:45.000618+00:00 end=None details=None />,
+    <Event type=CONJUNCTION objects=[<Object type=SATELLITE name=MOON />, <Object type=PLANET name=VENUS />] start=2020-01-28 11:01:51.909499+00:00 end=None details=None />,
+    <Event type=APOGEE objects=[<Object type=SATELLITE name=MOON />] start=2020-01-29 21:32:13.884314+00:00 end=None details={'distance_km': 405426.4150890029} />]
+
+    Find Apogee events between January 27th, 2020 and January 29th, 2020:
+
+    >>> search_events([EventType.APOGEE], date(2020, 1, 29), date(2020, 1, 27))
+    [<Event type=APOGEE objects=[<Object type=SATELLITE name=MOON />] start=2020-01-29 21:32:13.884314+00:00 end=None details={'distance_km': 405426.4150890029} />]
+
+    Find Apogee events between January 27th, 2020 and January 29th, 2020 (show times in UTC-6):
+
+    >>> search_events([EventType.APOGEE], date(2020, 1, 29), date(2020, 1, 27), -6)
+    [<Event type=APOGEE objects=[<Object type=SATELLITE name=MOON />] start=2020-01-29 15:32:13.884314-06:00 end=None details={'distance_km': 405426.4150890029} />]
+
+    If no events occurred in the given time range, an empty list is returned.
+
+    >>> event_types = [EventType.OPPOSITION, EventType.CONJUNCTION, EventType.SEASON_CHANGE, EventType.LUNAR_ECLIPSE]
+    >>> search_events(event_types, date(2021, 5, 15), date(2021, 5, 14))
+    []
+
+    Note that the events can only be found for a date range.
+    Asking for the events with an out of range date will result in an exception:
+
+    >>> event_types = [EventType.OPPOSITION, EventType.CONJUNCTION]
+    >>> search_events(event_types, date(1000, 1, 2), date(1000, 1, 1))
+    Traceback (most recent call last):
+        ...
+    kosmorrolib.exceptions.OutOfRangeDateError: The date must be between 1899-07-28 and 2053-10-08
+    """
+
+    def search_all_apogee_events(start: date, end: date, timezone: int = 0) -> [Event]:
+        moon_apogee_events = _search_apogee(ASTERS[1])(start, end, timezone)
+        earth_apogee_events = _search_apogee(EARTH, from_aster=ASTERS[0])(
```

Comment:
Can you set variables for `ASTERS[1]` (Moon) and `ASTERS[0]` (Sun) for better readability?